Mobile Buy SDK/Mobile Buy SDK Tests/BUYClientTest_Customer.m

```diff
@@ -153,4 +142,94 @@ - (void)testCustomerLogin
 	}];
 }
 
+- (void)testCustomerLogout
+{
+	[OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest * _Nonnull request) {
+		return [self shouldUseMocks];
+	} withStubResponse:^OHHTTPStubsResponse * _Nonnull(NSURLRequest * _Nonnull request) {
+		return [OHHTTPStubsResponse responseWithKey:@"testCustomerLogin"];
+	}];
+	
+	XCTestExpectation *expectation = [self expectationWithDescription:NSStringFromSelector(_cmd)];
+	[self.client loginCustomerWithCredentials:[self credentialsForLogin] callback:^(BUYCustomer *customer, NSString *token, NSError *error) {
+		
+		[OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest * _Nonnull request) {
+			return [self shouldUseMocks];
+		} withStubResponse:^OHHTTPStubsResponse * _Nonnull(NSURLRequest * _Nonnull request) {
+			return [OHHTTPStubsResponse responseWithKey:@"testCustomerLogout"];
+		}];
+		
+		[self.client logoutCustomerID:customer.identifier.stringValue callback:^(BUYStatus status, NSError * _Nullable error) {
+			
+			XCTAssertNil(error);
+			XCTAssertEqual(status, 204);
+			
+			[expectation fulfill];
+		}];
+	}];
+	
+	[self waitForExpectationsWithTimeout:10 handler:nil];
+}
+
+#pragma mark - Update -
+
+- (void)testCustomerUpdate
+{
+	[OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest * _Nonnull request) {
+		return [self shouldUseMocks];
+	} withStubResponse:^OHHTTPStubsResponse * _Nonnull(NSURLRequest * _Nonnull request) {
+		return [OHHTTPStubsResponse responseWithKey:@"testCustomerLogin"];
+	}];
+	
+	XCTestExpectation *expectation = [self expectationWithDescription:NSStringFromSelector(_cmd)];
+	
+	BUYAccountCredentialItem *email    = [BUYAccountCredentialItem itemWithEmail:self.customerEmail];
+	BUYAccountCredentials *credentials = [BUYAccountCredentials credentialsWithItems:@[email]];
+	
+	[self.client loginCustomerWithCredentials:[self credentialsForLogin] callback:^(BUYCustomer * _Nullable customer, NSString * _Nullable token, NSError * _Nullable error) {
+		
+		[OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest * _Nonnull request) {
```

Comment:
Can this stub request be added earlier than here?

I don't like how all these mocking code chunks noisy up the core test logic.

I wonder if it would be better, in the long run, to install all stubs in `+setup`, and define specific tests per URL, instead of using `-shouldUseMocks` in the test (but only install the stubs if `-shouldUseMocks` is `YES`.